mobile/types.go

```diff
@@ -264,8 +265,11 @@ func (tx *Transaction) GetHash() *Hash    { return &Hash{tx.tx.Hash()} }
 func (tx *Transaction) GetSigHash() *Hash { return &Hash{tx.tx.SigHash(types.HomesteadSigner{})} }
 func (tx *Transaction) GetCost() *BigInt  { return &BigInt{tx.tx.Cost()} }
 
-func (tx *Transaction) GetFrom() (address *Address, _ error) {
-	from, err := types.Sender(types.HomesteadSigner{}, tx.tx)
+func (tx *Transaction) GetFrom(chainID *big.Int) (address *Address, _ error) {
```

Comment:
Almost :)

The `mobile` package is a special little snowflake. APIs cannot contain fancy types, only primitives, or structs that expose only primitives. This is because the types need to cross Go -> C -> Java. Unfortunately this rules out having `big.Int` as a parameter.

The way the `mobile` package solves this is that all complex types that are needed in the API are actually wrapped inside the package. For `big.Int`, theres `BigInt`. E.g. the last parameter of `SignTx` https://github.com/ethereum/go-ethereum/blob/master/mobile/accounts.go#L118 and the `null` handling there. Please rewrite this snippet to use that.